src/cmdstan-guide/parallelization.Rmd

```diff
@@ -1,37 +1,173 @@
+`STAN_NUM_THREADS`. On Unix systems we can set it using the `export` command. The 
+following would set the maximum number of threads to use for a chain to four:
 ```
 export STAN_NUM_THREADS=4
-./parallel_logistic sample data file=data.csv ...
+```
+In Windows PowerShell use the following command:
+```
+Set-Variable -Name STAN_NUM_THREADS -Value "4"
+```
+
+Generally, this number should not exceed the number of available cores.
+If this variable isn't set, then the program will run with only one thread.
+
+Once `STAN_NUM_THREADS` is set, run a model as normal:
+```
+./model sample data file=data.json ...
+```
+
+## Multi-processing with MPI
+
+In order to use multi-processing with MPI in a Stan model, the models must be 
+rewritten to use the `map_rect` functions. By using MPI, the model can be 
+parallelized across multiple cores or a cluster. MPI with Stan is supported 
+on MacOS and Linux.
+
+### Dependencies
+
+Compiling and running Stan models with MPI requires that the system 
+has an MPI implementation installed. For Unix systems the most commonly used
+implementations are [MPICH](https://www.mpich.org/) and [OpenMPI](https://www.open-mpi.org/).
+
+### Compiling
+
+Once a model is rewritten to use `map_rect`, additional makefile flags 
+must be written to the `make/local`. These are:
+
+- `STAN_MPI`: Enables the use of MPI with Stan if `TRUE`.
+- `CXX`: The name of the MPI C++ compiler wrapper. Typically `"mpicxx"`.
+- `TBB_CXX_TYPE`: The C++ compiler the MPI wrapper wraps. Typically `"gcc"` on Linux and `"clang"` on macOS.
+
+An example of `make/local` on Linux:
+
+```
+STAN_MPI=true
+CXX=mpicxx
+TBB_CXX_TYPE=gcc
+```
+The model is then compiled as normal:
+```
+make path/to/model
+```
+### Running
+
+The Stan model compiled with `STAN_MPI` is then run using an MPI launcher. The MPI standard
+suggests using `mpiexec`, but a vendor wrapper for the launcher like `mpirun` can also be used.
+The launcher is supplied the path to the built executable and the number of processes to start:
+`-n X` for `mpiexec` or `-np X` for `mpirun` where `X` is replaced by the integer representing
+the number of processes.
+
+Example for running a model with six processes:
+```
+mpiexec -n 6 path/to/model sample data file=data.json ...
 ```
 
+## OpenCL
 
+### Dependencies
+
+OpenCL is supported on most modern CPUs and GPUs. In order to run OpenCL-enabled Stan models,
+an OpenCL runtime for the target device must be installed. This subsection lists installation
+instructions for OpenCL runtimes of the commonly-found devices.
+
+In order to check if any OpenCL-enabled device and its runtime is already present use the
+`clinfo` tool. On Linux, `clinfo` can typically be installed with the default package manager
+(for example `sudo apt-get install clinfo` on Ubuntu). For Windows, pre-built `clinfo` binary
+can be found [here](https://github.com/Oblomov/clinfo#windows-support).
```

Comment:
Is this really the clinfo Windows users use? Is there not a Microsoft one or something? From the Github: "The application can usually be built in Windows too"

Reply:
Yes, this is the Windows clinfo equivalent. 

There are all sorts of heavyweight GUI apps that also list the IDs, but this is a single exe file and straightforward.

Reply:
You can download the built executable, which is what this links too (bottom of README).